pkg/backporter/backporter.go

```diff
@@ -0,0 +1,292 @@
+	<p> Target Release: {{ .Bug.TargetRelease }} </p>
+	{{ if .PRs }}
+		<p> GitHub PR: 
+		{{ if .PRs }}
+			{{ range $index, $pr := .PRs }}
+				{{ if $index}}|{{end}} 
+				<a href="{{ $pr.Type.URL }}/{{ $pr.Org}}/{{ $pr.Repo}}/pull/{{ $pr.Num}}">{{ $pr.Org}}/{{ $pr.Repo}}#{{ $pr.Num}}</a>
+			{{ end }}
+		{{ end }}
+		</p>
+	{{ else }}
+		<p> No linked PRs! </p>
+	{{ end }}
+	{{ if ne .Parent.ID .Bug.ID}}
+		<p> Cloned From: <a href = "/getclones?ID={{.Parent.ID}}"> Bug {{.Parent.ID}}: {{.Parent.Summary}}</a> | Status: {{.Parent.Status}}
+	{{ else }}
+		<p> Cloned From: This is the original! </p>
+	{{ end }}
+	<h4 id="clones"> <a href ="#clones"> Clones</a> </h4>
+	<table class="table">
+		<thead>
+			<tr>
+				<th title="Targeted version to release fix" class="info">Target Release</th>
+				<th title="ID of the cloned bug" class="info">Bug ID</th>
+				<th title="Status of the cloned bug" class="info">Status</th>
+				<th title="PR associated with this bug" class="info">PR</th>
+			</tr>
+		</thead>
+		<tbody>
+		{{ if .Clones }}
+			{{ range $clone := .Clones }}
+				<tr>
+					<td style="vertical-align: middle;">{{ $clone.TargetRelease }}</td>
+					<td style="vertical-align: middle;"><a href = "/getclones?ID={{$clone.ID}}">{{ $clone.ID }}</a></td>
+					<td style="vertical-align: middle;">{{ $clone.Status }}</td>
+					<td style="vertical-align: middle;">
+						{{range $index, $pr := $clone.PRs }}
+							{{ if $index}},{{end}}
+							<a href = "{{ $pr.Type.URL }}/{{$pr.Org}}/{{$pr.Repo}}/pull/{{$pr.Num}}" target="_blank"> {{$pr.Org}}/{{$pr.Repo}}#{{$pr.Num}}</a>
+						{{end}}
+					</td>
+				</tr>
+			{{ end }}
+		{{ else }}
+			<tr> <td colspan=4 style="text-align:center;"> No clones found! </td></tr>
+		{{ end }}
+		</tbody>
+	</table>`
+
+var (
+	clonesTemplate = template.Must(template.New("clones").Parse(clonesTemplateConstructor))
+	emptyTemplate  = template.Must(template.New("empty").Parse("{{.}}"))
+)
+
+// HandlerFuncWithErrorReturn allows returning errors to be logged
+type HandlerFuncWithErrorReturn func(http.ResponseWriter, *http.Request) error
+
+// Class to hold the UI data for the clones page
+type clonesTemplateData struct {
+	Bug    *bugzilla.Bug          //bug details
+	Clones []*bugzilla.Bug        // List of clones for the bug
+	Parent *bugzilla.Bug          // Root bug if it is a a bug, otherwise holds itself
+	PRs    []bugzilla.ExternalBug // Details of linked PR
+}
+
+//Writes an HTML page, prepends header in htmlPageStart and appends header from htmlPageEnd around tConstructor.
+func writePage(w http.ResponseWriter, title string, body *template.Template, data interface{}) error {
+	fmt.Fprintf(w, htmlPageStart, title)
+
+	if err := body.Execute(w, data); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "%s: %v", http.StatusText(http.StatusInternalServerError), err)
+		return err
+	}
+	fmt.Fprint(w, htmlPageEnd)
+	return nil
```

Comment:
I am not a fan of ignoring the errors from the `Fprintf` here - `writePage is supposed to return an `error` when something goes wrong, why not propagate (or at least log) them?

Reply:
qq - Is this comment specific to this Fprintf or do you mean that I should have done that in general?

Reply:
Ok. I have added error checks inside the error handlers.

Reply:
Still missing an errcheck right here :upside_down_face:

Reply:
The first time I checked I somehow didnt see that fprint returned an error, which caused this whole thread. I corrected that mistake😓. Thank you for pointing it out